### ISSUE TITLE
Minor bug: The use of 'expected' instead of `unexpected` for indent violations.

### ISSUE THREAD

**Issue by lozandier** [eslint/eslint#3220]
For violations of the indent rule, the error message has a typo that makes it seem that the user's incorrect indentation is correct because `Unexpected` is used instead of `Expected` to describe the violation: 
<img width="545" alt="screen shot 2015-07-31 at 12 49 42 am" src="https://cloud.githubusercontent.com/assets/1223224/9006766/14705d24-3759-11e5-9cea-0ee22e0145c6.png">

This *may* be related to #3173, but this also occurs with control structures, & etc. in a way that seems more like a typo with the error message

## Relevant rules
Here's the rules I'm using to get this behavior:
```js
"indent": [
  2,
  2,
],
```

## Other Relevant Information
- ESLint version `v0.24.1`

## Entire Config Object
```js

const ESLintConfig = {
    "rules": {
        "indent": [
            2,
            2,
        ],

        "dot-location":  [
          2,
          "object"
        ],

        "no-return-assign": [
          2
        ],

        "camelcase": [
          2,
          {
            properties: "always"
          },
        ],

        "no-inline-comments": [
          2,
        ],

        "max-nested-callbacks": [
          2,
          3,
        ],

        "linebreak-style": [
            2,
            "unix",
        ],

        "semi": [
            2,
            "always",
        ],

        "use-isnan": [
          2,
        ],

        "space-in-parens": [
          2,
          "always",
          {
            "exceptions": [ "empty", "[]" ],
          }
        ],

        "valid-jsdoc": [
          2,
        ],

        "no-mixed-spaces-and-tabs": [
          2,
        ],

        "quotes": [
          1,
          "backtick"
        ],

        "no-extra-semi": [
          2,
        ],

        "no-var": [
          1,
        ],

        "no-this-before-super": [
          2,
        ],

        "prefer-const": [
          1
        ],

        "computed-property-spacing": [
          2,
          "always",
        ],

        "object-shorthand": [
          2,
          "always",
        ],

        "constructor-super": [
          2,
        ],

        "padded-blocks": [
          2,
          "always",
        ],

        "space-before-function-paren": [
          2,
          "never",
        ],

        "no-multiple-empty-lines": [
          2,
          {
            max: 2,
          }
        ],

        "comma-spacing": [
          2,
          {
            before: false,
            after: true,
          }
        ],

        "no-spaced-func": [
          2
        ],

        "comma-style": [
          2,
          "first",
        ],

        "new-cap": [
          2,
        ],

        "no-trailing-spaces": [
          2,
          {
            skipBlankLines: true,
          }
        ],

        "spaced-comment": [
          2,
          "always",
          {
            exceptions: [ "-" ],
            markers: ["="],
          }
        ],

        "space-infix-ops": [
          2,
          {
            int32Hint: true,
          }
        ],

        "func-style": [
          2,
          "expression",
        ],

        "array-bracket-spacing": [
          2,
          "always",
          {
            "singleValue": false,
            "objectsInArrays": false,
            "arraysInArrays": false,
          }
        ],

        "quote-props": [
          2,
          "as-needed",
        ],

        "valid-typeof": [
          2,
        ],

        "no-control-regex": [
          2,
        ],

        "no-regex-spaces": [
          2,
        ],

        "no-empty": [
          2,
        ],

        "no-extra-boolean-cast": [
          2,
        ],

        "no-dupe-keys": [
          2,
        ],

        "no-negated-in-lhs": [
          2,
        ],

        "no-obj-calls": [
          2,
        ],

        "no-ex-assign": [
          2,
        ],

        "no-unreachable": [
          2,
        ],

        "no-constant-condition": [
          2,
        ],

        "no-sparse-arrays": [
          2,
        ],

        "no-unexpected-multiline": [
          2,
        ],

        "no-dupe-args": [
          2,
        ],

        "no-cond-assign": [
          2,
        ],

        "no-unused-vars": [
          1,
        ],

        "vars-on-top": [
          2,
        ],

        "no-new-wrappers": [
          2,
        ],

        "no-eval": [
          2,
        ],

        "no-multi-spaces": [
          2,
        ],

        "eqeqeq": [
          2,
          "smart"
        ],

        "no-script-url": [
          2,
        ],

        "no-throw-literal": [
          2,
        ],

        "comma-dangle": [
          2,
          "always-multiline",
        ],

        "strict": [
          2,
          "global"
        ],

        "global-strict" :[
          0
        ],

    },
    "env": {
        "es6": true,
        "browser": true,
        "mocha": true
    },

    "globals": {
      "dialogPolyfill": true,
      "$": true,
      "jQuery": true
    }
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by IanVS** [comment]
> Unexpected is used instead of Expected

In your screengrab of the error, the message indeed uses `Expected`, as it should.  Can you describe the issue a little more?

**Comment by lozandier** [comment]
@ianVS  Maybe I'm interpreting the error message wrong this whole time, Yeah, I thought it would say something along the lines of `Unexpected indentation of 8 characters` as `Expected` sounded to me that I wanted an indentation of 8 characters instead of 2. 

Maybe I implicitly thought the error message would say something along the lines of being clear on *how* to fix the line complying with an indentation level of a specific amount set by the `rules` object (or explicitly pointing out what that is vs what was done).

**Comment by IanVS** [comment]
You should be able to tell how many characters of indentation you have by looking at your code, so telling you in the message how many are there would be less-than-useful.  Instead, the message is telling you how many there should be, so it's a simple matter of checking your code, and putting in the number of characters that are expected.

If this is still confusing, maybe if you can share a simple block of code and rule message, and we can talk through it with code.

**Comment by gyandeeps** [comment]
I agree with @IanVS . With this style it helps me to fix the code by telling me the expected indent.

**Comment by nzakas** [comment]
"Expected x but found y." Is a really nice template for error messages. If we can easily add " but found 12" at the end, I think we should do it.

**Comment by gyandeeps** [comment]
Thats pretty simple as I am already calculating what we have vs expected. It just I dont display it.
Should I do it with the existing PR?
